samples/cellular/modem_shell/src/sock/sock.c

```diff
@@ -992,8 +999,14 @@ int sock_send_data(
 		}
 
 	} else if (data_out != NULL && data_out_length > 0) {
+		/* Set requested blocking mode for duration of data sending */
+		set_sock_blocking_mode(socket_info->fd, blocking);
```

Comment:
For periodic sending (`-e` option) it's possible to subtract the time used in `send()` before scheduling next send. A small corner case if setting `-e 1` and the send uses more than 1 second, but I have only seen this happen on wakeup from PSM.

For sending a chunk of data (`-d` option) I think it's good to be able to specify blocking or non-blocking. Maybe also printing the time used in send when having the `-W` option.

Reply:
This seems to be my misunderstanding. So adding blocking mode handling for `-d` option was not directly related to `-W` option, I guess. I somehow thought it was needed to make `-W` option work at all.

> For periodic sending (`-e` option) it's possible to subtract the time used in `send()` before scheduling next send. A small corner case if setting `-e 1` and the send uses more than 1 second, but I have only seen this happen on wakeup from PSM.

I'm not sure if we need to subtract anything or just let the interval start when sending is complete, i.e., how it should work with the current code.

> For sending a chunk of data (`-d` option) I think it's good to be able to specify blocking or non-blocking.

True.

> Maybe also printing the time used in send when having the `-W` option.

Could be done but not necessary.

Reply:
> I made blocking mode configurable for `-d` to be able to test different combinations, but maybe we should print a warning in case `-B 0 -W` is used?

I guess it would be good to avoid unintentional misuse.

Reply:
I documented this restriction.